week9/answers.sql

```diff
@@ -1,3 +1,77 @@
 -- Create your tables, views, functions and procedures here!
 CREATE SCHEMA social;
 USE social;
+
+CREATE TABLE users (
+  user_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  first_name VARCHAR(30) NOT NULL,
+  last_name VARCHAR(30) NOT NULL,
+  email VARCHAR(50) NOT NULL,
+  created_on TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+CREATE TABLE sessions (
+  session_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  user_id INT UNSIGNED NOT NULL,
+  created_on TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_on TIMESTAMP NOT NULL DEFAULT NOW() ON UPDATE NOW()
+  CONSTRAINT sessions_fk_users
+    FOREIGN KEY (user_id)
+    REFERENCES users (user_id)
+    ON UPDATE CASCADE
+    ON DELETE CASCADE
+);
+CREATE TABLE friends (
+  user_friend_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  user_id INT UNSIGNED NOT NULL,
+  friend_id INT UNSIGNED NOT NULL
+  CONSTRAINT friends_fk_users
+    FOREIGN KEY (user_id)
+    REFERENCES users (user_id)
+    ON UPDATE CASCADE
+    ON DELETE CASCADE,
+  CONSTRAINT friends_fk_friend
+    FOREIGN KEY (friend_id)
+    REFERENCES users (user_id)
+    ON UPDATE CASCADE
+    ON DELETE CASCADE
+);
+
+CREATE TABLE posts (
+  post_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  user_id INT UNSIGNED NOT NULL,
+  created_on TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_on TIMESTAMP NOT NULL DEFAULT NOW() ON UPDATE NOW(),
+  content VARCHAR(140) NOT NULL
+  CONSTRAINT posts_fk_users
+    FOREIGN KEY (user_id)
+    REFERENCES users (user_id)
+    ON UPDATE CASCADE
+    ON DELETE CASCADE,
+);
+
+CREATE TABLE notifications (
+  notification_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+  user_id INT UNSIGNED NOT NULL,
+  post_id INT UNSIGNED NOT NULL
+  CONSTRAINT notifications_fk_users
+    FOREIGN KEY (user_id)
+    REFERENCES users (user_id)
+    ON UPDATE CASCADE
+    ON DELETE CASCADE,
+  CONSTRAINT notifications_fk_posts
+    FOREIGN KEY (post_id)
+    REFERENCES posts (post_id)
+    ON UPDATE CASCADE
+    ON DELETE CASCADE,
+);
+
+CREATE OR REPLACE VIEW notification_posts AS
+  SELECT notifications.user_id AS user_id, users.first_name AS first_name, users.last_name AS last_name, posts.post_id AS post_id, posts.content AS content
+    FROM notifications      
```

Comment:
REALLY close. Some notes:

- INNER JOIN -> LEFT OUTER JOIN
- Thinking about how to join these tables... you properly started with `notifications`, but then you'll want to bring in the `posts` table, then the `users` table to get the user that created the post.